maro/rl_v3/training/algorithms/maddpg.py

```diff
@@ -363,26 +367,24 @@ def record(self, exp_element: ExpElement) -> None:
     def _get_batch(self, batch_size: int = None) -> MultiTransitionBatch:
         return self._replay_memory.sample(batch_size if batch_size is not None else self._batch_size)
 
-    def _get_local_ops_by_name(self, ops_name: str) -> AbsTrainOps:
+    def get_local_ops_by_name(self, ops_name: str) -> AbsTrainOps:
         if ops_name == self._shared_critic_ops_name:
-            params = {
-                **self._params.extract_ops_params(),
-                "get_policy_func": None,
-                "policy_idx": -1,
-                "shared_critic": False,
-                "create_actor": False,
-            }
-            return DiscreteMADDPGOps(**params)
+            return DiscreteMADDPGOps(
+                get_policy_func=None,
+                policy_idx=-1,
+                shared_critic=False,
+                create_actor=False,
+                **self._ops_params
```

Comment:
`self._ops_params` should be put at the first place, and let the following parameters to overwrite. For example, `shared_critic`.